File: mahjong-view/src/app/model/player-score-dto.ts
export class PlayerScoreDto {

    playerId: number;
    combinationCodes: string[];
    doraCount: number;
    fuCount: number;
    openHand: boolean;
    riichi: boolean;
    tempai: boolean;

    static fromTempai(playerId: number, tempai: boolean, riichi: boolean): PlayerScoreDto {
        let dto = this.fromRiichi(playerId, riichi);
        dto.tempai = tempai;
        return dto;
    }

    static fromWin(playerId: number,
                   combinations: string[],
                   riichi: boolean,
                   doraCount: number,
                   fuCount: number,
                   openHand: boolean): PlayerScoreDto {
        let dto = this.fromRiichi(playerId, riichi);
        dto.combinationCodes = combinations;
        dto.doraCount = doraCount;
        dto.fuCount = fuCount;
        dto.openHand = openHand;
        return dto;
    }

    static fromRiichi(playerId: number, riichi: boolean): PlayerScoreDto {
        let dto = new PlayerScoreDto();
        dto.playerId = playerId;
        dto.combinationCodes = [];
        dto.doraCount = 0;
        dto.fuCount = 0;
        dto.openHand = false;
        dto.riichi = riichi;
        dto.tempai = false;
        return dto;
    }

}

File: mahjong-main-public-api/src/main/java/com/github/mahjong/main/publicapi/dto/PlayerScoreDto.java
package com.github.mahjong.main.publicapi.dto;

import javax.validation.constraints.NotNull;
import java.util.ArrayList;
import java.util.List;

public class PlayerScoreDto {

    /**
     * Id of player to whom this score belongs to
     */
    public long playerId;
    /**
     * List of combination codes, that player collect in his hand
     */
    @NotNull
    public List<String> combinationCodes = new ArrayList<>();
    /**
     * Count of doras
     */
    public int doraCount;
    /**
     * Amount of fu points
     */
    public int fuCount;
    /**
     * True if player opened his hand
     */
    public boolean openHand;
    /**
     * True if player has riichi and game's rules set support riichi combination
     */
    public boolean riichi;
    /**
     * True if player DOESN'T win, but has a tempai at the end (if it make sense)
     */
    public boolean tempai;
}
